pkgs/development/libraries/rocfft/default.nix

```diff
@@ -1,199 +1,246 @@
-  name-one = "librocfft-device-1.so.0.1";
-  name-two = "librocfft-device-2.so.0.1";
-  name-three = "librocfft-device-3.so.0.1";
-
-  # This is over 3GB, to allow hydra caching we separate it
-  rf = stdenv.mkDerivation (finalAttrs: {
-    pname = "rocfft";
-    version = "5.4.3";
-
-    outputs = [
-      "out"
-      "libzero"
-      "libone"
-      "libtwo"
-      "libthree"
-    ] ++ lib.optionals buildTests [
-      "test"
-    ] ++ lib.optionals buildBenchmarks [
-      "benchmark"
-    ];
-
-    src = fetchFromGitHub {
-      owner = "ROCmSoftwarePlatform";
-      repo = "rocFFT";
-      rev = "rocm-${finalAttrs.version}";
-      hash = "sha256-FsefE0B2hF5ZcHDB6TscwFeZ1NKFkWX7VDpEvvbDbOk=";
-    };
-
-    nativeBuildInputs = [
-      cmake
-      rocm-cmake
-      hip
-    ];
-
-    buildInputs = [
-      sqlite
-      python3
-    ] ++ lib.optionals buildTests [
-      gtest
-    ] ++ lib.optionals (buildTests || buildBenchmarks) [
-      rocrand
-      boost
-      fftw
-      fftwFloat
-      openmp
-    ];
-
-    propagatedBuildInputs = lib.optionals buildTests [
-      fftw
-      fftwFloat
-    ];
-
-    cmakeFlags = [
-      "-DCMAKE_C_COMPILER=hipcc"
-      "-DCMAKE_CXX_COMPILER=hipcc"
-      "-DUSE_HIP_CLANG=ON"
-      "-DSQLITE_USE_SYSTEM_PACKAGE=ON"
-      # Manually define CMAKE_INSTALL_<DIR>
-      # See: https://github.com/NixOS/nixpkgs/pull/197838
-      "-DCMAKE_INSTALL_BINDIR=bin"
-      "-DCMAKE_INSTALL_LIBDIR=lib"
-      "-DCMAKE_INSTALL_INCLUDEDIR=include"
-    ] ++ lib.optionals buildTests [
-      "-DBUILD_CLIENTS_TESTS=ON"
-    ] ++ lib.optionals buildBenchmarks [
-      "-DBUILD_CLIENTS_RIDER=ON"
-      "-DBUILD_CLIENTS_SAMPLES=ON"
-    ];
-
-    postInstall = ''
-      mv $out/lib/${name-zero} $libzero
-      mv $out/lib/${name-one} $libone
-      mv $out/lib/${name-two} $libtwo
-      mv $out/lib/${name-three} $libthree
-      ln -s $libzero $out/lib/${name-zero}
-      ln -s $libone $out/lib/${name-one}
-      ln -s $libtwo $out/lib/${name-two}
-      ln -s $libthree $out/lib/${name-three}
-    '' + lib.optionalString buildTests ''
-      mkdir -p $test/{bin,lib/fftw}
-      cp -a $out/bin/* $test/bin
-      ln -s ${fftw}/lib/libfftw*.so $test/lib/fftw
-      ln -s ${fftwFloat}/lib/libfftw*.so $test/lib/fftw
-      rm -r $out/lib/fftw
-      rm $test/bin/{rocfft_rtc_helper,*-rider} || true
-    '' + lib.optionalString buildBenchmarks ''
-      mkdir -p $benchmark/bin
-      cp -a $out/bin/* $benchmark/bin
-      rm $benchmark/bin/{rocfft_rtc_helper,*-test} || true
-    '' + lib.optionalString (buildTests || buildBenchmarks ) ''
-      mv $out/bin/rocfft_rtc_helper $out
-      rm -r $out/bin/*
-      mv $out/rocfft_rtc_helper $out/bin
+  # To avoid output limit exceeded errors in hydra, we build kernel
+  # device libs and the kernel RTC cache database in separate derivations
+  #
+  # TODO: Figure out why parallel linking isn't working with hipcc
+  # (-fuse-ld=lld is used by default in rocmClangStdenv)
```

Comment:
I haven't really figured out why the linker doesn't run in parallel - but I don't think it should block this PR since this PR will fix the hydra build.

Reply:
I’m not sure about the context of what the parallel linking here is, but there’s a couple of things that could happen:
1. If it is GPU code, it’s probably forced to use LTO, i.e. the compiler writes its IR into the .o files and the linker merges all the IR and then compiles it all at once. (Function calls and linking are hard on GPUs, so LTO is or at least was used, I’m not perfectly up-to-date on the state in hip.)
2. If it’s CPU code involving LLVM, linking is often limited to just a couple threads because you run out of RAM when using ld.bfd and compiling in debug mode.

Reply:
Ok thanks! It's probably forced to use LTO then.